src/ttt/net/two_headed_net.rs

```diff
@@ -22,7 +22,7 @@ impl TwoHeadedNet {
     }
 }
 
-impl ValueFunction<TttGame> for TwoHeadedNet {
+impl<const CPU: bool> ValueFunction<TttGame> for TwoHeadedNet<CPU> {

```

Comment:
I guess that can work.
I kind of like the use of generics here, as all the 'if's are known at compile time.
Maybe we can get the env variable at the rust main and propagate it from there using generics.
But in general i would like to avoid env variable, its hard to follow who set them and where are they used

Reply:
If we propagate in the same way then I don't see a reason to use env variable instead of argument. Yes I agree about compile time coolness, and that generally passing information using env variables is confusing, but in this case it really is a potentially global trait of the system (has or doesn't have GPU).
Kitzur if you prefer the current method then let's keep it, both seem fine to me